src/components/graph-sonification.tsx

```diff
@@ -22,153 +52,284 @@ export const GraphSonification = ({ graphToSonify, onResetGraphToSonify }: IProp
-          });
-        }, offsetSeconds);
-      });
+    onSelectGraph(graphDetails);
+  };
 
-      Tone.getTransport().start();
-      const step = () => {
-        const elapsed = Tone.getTransport().seconds;
-        const fraction = Math.min(elapsed / totalDuration, 1);
-
-        codapInterface.sendRequest({
-          action: "update",
-          resource: `component[${graphDetails.id}].adornment`,
-          values: {
-            type: "Region of Interest",
-            primary: { "position": `${fraction * 100}%`, "extent": 0.05 }
-          }
-        });
+  useEffect(() => {
+    isLoopingRef.current = isLooping;
+  }, [isLooping]);
 
-        if (fraction < 1) {
-          frameIdRef.current = requestAnimationFrame(step);
-        } else {
-          if (frameIdRef.current) {
-            cancelAnimationFrame(frameIdRef.current);
-            frameIdRef.current = null;
-          }
-          codapInterface.sendRequest({
-            action: "update",
-            resource: `component[${graphDetails.id}].adornment`,
-            values: {
-              type: "Region of Interest",
-              primary: { "position": "0%", "extent": 0 }
-            }
-          });
-          Tone.getTransport().stop();
-          onResetGraphToSonify();
-        }
-      };
-  
-      frameIdRef.current = requestAnimationFrame(step);
-    };
+  useEffect(() => {
+    durationRef.current = kDefaultDuration / speed;
+  }, [speed]);
 
-    fetchGraphData();
+  useEffect(() => {
+    // Since the selected graph can be set outside the component (i.e. not using the component's graph select menu),
+    // we call `mapValuesToTimeAndPitch` in an effect.
+    if (selectedGraph) {
+      mapValuesToTimeAndPitch(selectedGraph);
+    }
+  }, [selectedGraph]);
 
+  useEffect(() => {
     return () => {
+      // Ensure requestAnimationFrame is cancelled if the component unmounts during playback
       if (frameIdRef.current) {
         cancelAnimationFrame(frameIdRef.current);
         frameIdRef.current = null;
       }
-      Tone.getTransport().stop();
-
-      if (currentGraphIdRef.current) {
-        removeROIAdornment(currentGraphIdRef.current);
-        currentGraphIdRef.current = null;
-      }
     };
-  }, [graphToSonify, onResetGraphToSonify]);
+  }, []);
 
   return (
-    <div hidden={true}/>
+    <div className="graph-sonification control-panel" role="group" aria-labelledby="control-panel-heading">
+      <h2 id="control-panel-heading">Sonification</h2>
+      <div className="graph-selection">
+        <label htmlFor="graph-select">Graph to sonify:</label>
+        <select id="graph-select" value={selectedGraph?.id || ""} onChange={(e) => handleSelectGraph(e.target.value)}>
+          <option value="" disabled>Select a graph</option>
+          {availableGraphs.map((graph) => (
+            <option key={graph.id} value={graph.id}>
+              {graph.name || graph.title || graph.id}
+            </option>
```

Comment:
It seems like the displayed name here ought to match what is shown in the title bar of the graph. But when I create my own graph it shows "Mammals" in the interface, but an ID number in this pulldown.

Reply:
This happens because even when you create a new graph in CODAP it has no 'name' or 'title' property, it just displays the dataset name in its component title bar.
I've changed the plugin code for now so that the dataset name appears in the dropdown list with an index if there are other graphs without titles/names that also display the dataset in their title bar.
Note this is a potentially temporary solution for now, because there's more thinking / communicating to be done between the DAVAI and CODAP teams about how to best resolve this issue.